src/main/java/org/apache/sling/jcr/resource/internal/helper/JcrPropertyMapCacheEntry.java

```diff
@@ -132,7 +132,7 @@ private static void failIfCannotStore(final @NotNull Object value, final @NotNul
                 final ByteArrayInputStream bais = new ByteArrayInputStream(baos.toByteArray());
                 value = session.getValueFactory().createValue(session.getValueFactory().createBinary(bais));
             } catch (IOException ioe) {
-                // we ignore this here and return null
+                throw new RepositoryException("Cannot serialize object", ioe);
```

Comment:
It would make sense to also log the node path and the type of the object in this exception, as it can help to further debug the issue in more detail.

Reply:
This is done in the the caller `JcrModificableValueMap.put(String, Object)` which puts a wrapping exception around this adding this info.